src/main/resources/xml/schema/2020-1/nordic-html5.rng

```diff
@@ -2088,55 +2087,66 @@
-            <!-- Strict content model: (h1 | span | p | section | img | figure)* -->
-            <ref name="attlist.poem"/> <!-- @epub:type, @class, @id, @title, @xml:space, @xml:lang, @lang, @dir -->
-            <zeroOrMore>
+        <element name="div">
+            <attribute name="class">
+                <value>verse</value>
+            </attribute>
+            <optional>
                 <choice>
                     <ref name="title"/> <!-- h1, h2, h3, h4, h5, h6 -->
-                    <ref name="author.block"/> <!-- p -->
-                    <ref name="hd"/> <!-- h1, h2, h3, h4, h5, h6 -->
-                    <ref name="dateline"/> <!-- span -->
-                    <ref name="epigraph"/> <!-- blockquote, p -->
-                    <ref name="byline"/> <!-- span -->
-                    <ref name="linegroup"/> <!-- div, section -->
-                    <ref name="line"/> <!-- p -->
-                    <ref name="pagebreak.block"/> <!-- div -->
-                    <ref name="img"/> <!-- img -->
-                    <ref name="figure.imggroup"/> <!-- figure -->
-                    <ref name="sidebar"/> <!-- aside, figure -->
+                    <ref name="bridgehead"/>
                 </choice>
-            </zeroOrMore>
+            </optional>
+            <ref name="poem.linegroup"/>
+            <optional>
+                <element name="p">
+                    <attribute name="class">
+                        <value>verse-author</value>
+                    </attribute>
+                </element>
+            </optional>
         </element>
     </define>
 
-    <define name="attlist.poem" combine="interleave">
-        <attribute name="epub:type">
-            <list>
-                <ref name="types"/>
+    <define name="poem.linegroup">
+        <element name="p">
+            <attribute name="class">
+                <value>linegroup</value>
+            </attribute>
+            <oneOrMore>
+                <ref name="poem.line"/>
+                <ref name="poem.linenum"/>
+            </oneOrMore>
+        </element>
+    </define>
+
+    <define name="poem.line">
+        <element name="span">
+            <attribute name="class">
                 <choice>
-                    <!-- By mistake, z:3998:verse is allowed on the poem in guidelines version 2015-1, so for now we'll allow suppliers to either follow the guidlines (z3998:verse) or do it correctly (z3998:poem) -->
-                    <value>z3998:verse</value>
-                    <value>z3998:poem</value>
+                    <value>line</value>
```

Comment:
In cases where we have some required classes, should we also open up for all other types of classes as well? For instance, what if we want to add a class during our production workflow that will make this invalid:

```html
<span class="line highlight">…</span> <!-- (just an example, we don't use this class in our books) -->
```

On the other hand, if we allow "line + any other class", we would allow things like:

```
<span class="line verse verse-author">…</span>
```

…so we would need to allow only things that are not already defined to have a special meaning. I'm not sure if that's possible in RelaxNG. Maybe it would be possible to refer to the "classes" definition, and then into the `<data type="string"/>` at the end, add a list of exclusions, which would be the list of predefined classes. It might be possible to do `<data type="string"><except><value>line</value><value>linegroup</value><!-- etc. --></except></data>`, but I haven't tested.

You don't need to change it in this PR, but maybe we should go through all class attributes and add support for that later (@martinpub and @AndersEkl might have opinions here as well).